mono/utils/mono-threads.h

```diff
@@ -464,7 +464,7 @@ mono_thread_info_describe_interrupt_token (THREAD_INFO_TYPE *info, GString *text
 gboolean
 mono_thread_info_is_live (THREAD_INFO_TYPE *info);
 
-HANDLE
+MonoThreadInfo*
 mono_threads_create_thread (MonoThreadStart start, gpointer arg, MonoThreadParm *tp, MonoNativeThreadId *out_tid);
 
```

Comment:
I think this is a bit more complicated, since MonoThreadInfo pointers only last until the thread exits, and they are kept alive in other threads by hazard pointers (see mono_thread_info_lookup), so they need to be handled with care in other threads.